wandb/sdk/internal/internal_api.py

```diff
@@ -2000,29 +2002,31 @@ def download_file(self, url: str) -> Tuple[int, requests.Response]:
     def download_write_file(
         self,
         metadata: Dict[str, str],
-        out_dir: Optional[str] = None,
-    ) -> Tuple[str, Optional[requests.Response]]:
+        out_dir: Optional[PathOrStr] = None,
+    ) -> Tuple[PathOrStr, Optional[requests.Response]]:
         """Download a file from a run and write it to wandb/.
 
         Arguments:
             metadata (obj): The metadata object for the file to download. Comes from Api.download_urls().
-            out_dir (str, optional): The directory to write the file to. Defaults to wandb/
+            out_dir (Union[str, os.PathLike], optional): The directory to write the file to. Defaults to wandb/
 
         Returns:
             A tuple of the file's local path and the streaming response. The streaming response is None if the file
             already existed and was up-to-date.
         """
-        filename = metadata["name"]
-        path = os.path.join(out_dir or self.settings("wandb_dir"), filename)
+        filename = Path(metadata["name"])
+        path: PathOrStr = Path(out_dir or self.settings("wandb_dir")) / filename
         if self.file_current(filename, B64MD5(metadata["md5"])):
-            return path, None
+            return FilePathStr(str(path)) if isinstance(out_dir, str) else path, None
 
         size, response = self.download_file(metadata["url"])
 
         with util.fsync_open(path, "wb") as file:
             for data in response.iter_content(chunk_size=1024):
                 file.write(data)
 
+        if isinstance(out_dir, str):
+            path = FilePathStr(str(path))
```

Comment:
i understand the purpose of this type, but it seems more cumbersome for minor benefit. what do you think?